tests/pds/dataservice_test.go

```diff
@@ -41,21 +41,21 @@ var _ = Describe("{DeletePDSPods}", func() {
 				})
 
 				Step("get pods from pds-system namespace", func() {
-					if len(deploymentPods) != 0 {
-						deploymentPods = nil
+					if len(pdsPods) != 0 {
+						pdsPods = nil
 					}
 					podList, err = pdslib.GetPods(pdsNamespace)
 					log.FailOnError(err, "Error while getting pods")
 					log.Infof("PDS System Pods")
 					for _, pod := range podList.Items {
 						log.Infof("%v", pod.Name)
-						deploymentPods = append(deploymentPods, pod)
+						pdsPods = append(pdsPods, pod)
 					}
```

Comment:
This is confusing, why don't we get podList first `pdslib.GetPods(pdsNamespace)`, and then append them to []pdsPods? there is no point to check `len(pdsPods)` if you haven't even got the list of them yet

Reply:
`pdsPods` is a global variable and its used by other tests also, I expect always `pdsPods []corev1.Pod` to be empty at the start of the test.

Reply:
You should just do this and define pdsPods afrer you get them and get rid of setting it to `nil`
```
					podList, err := pdslib.GetPods(pdsNamespace)
					log.FailOnError(err, "Error while getting pds pods")
					log.Infof("PDS system pods")
					for _, pod := range podList.Items {
						log.Infof("%v", pod.Name)
						pdsPods = append(pdsPods, pod)
					}
```

Reply:
Test `DeletePDSPods` runs in a for loop (depends on the data services we are passing as a param ) and each time it will populate `pdsPods`. 
For first iteration it will have set of pods say ["xxx", "yyy"] and it will delete those pods as per the test case and new pods will get created in different names ["aaa","bbb"]. In the next iteration we need to delete the new pods but  the `pdsPods` will have the old pod entries so we need to empty `pdsPods` else `pdsPods` will have entries like  ["xxx", "yyy", "aaa", "bbb"] and during pod deletion step the test will fail because pods ["xxx", "yyy"] will not be available. 
hope I have cleared the confusion here.

Reply:
I have removed that check